File: android/app/src/main/java/com/noop/testcentre/BatteryGuidedCapture.kt
package com.noop.testcentre

/**
 * Guided "wear it N days, then Report" capture for the Battery test mode (#713, Test Centre). Twin of the
 * Swift BatteryGuidedCapture. Arming reuses the daily scheduled export (the DebugExportScheduler settings)
 * so a redacted bundle drops each morning while the mode runs; the day count is derived from the
 * started-at timestamp against the registry default, so there is no extra persisted clock to drift. The
 * status formatter is pure (ms in) so it is JVM-unit-testable. No em-dashes.
 */
object BatteryGuidedCapture {

    private const val DAY_MS = 86_400_000L

    /** "Capturing day K of N" for the Test Centre Battery row. K = full-days-elapsed + 1, capped at N;
     *  once N whole days pass it reads complete. Pure (now injected) so it is unit-testable. */
    fun statusText(startedAtMs: Long?, target: Int, nowMs: Long): String {
        if (startedAtMs == null) return "Not started"
        val elapsedDays = ((nowMs - startedAtMs) / DAY_MS).toInt()
        if (elapsedDays >= target) return "Capture complete, $target of $target days"
        return "Capturing day ${elapsedDays + 1} of $target"
    }
}

File: Strand/System/BatteryGuidedCapture.swift
import Foundation
import StrandAnalytics

// BatteryGuidedCapture.swift - the guided "wear it N days, then Report" flow for the Battery test mode
// (#713, Test Centre). The days-twin of the Sleep nights flow (GuidedCapture / GuidedCaptureProgress).
// Arming reuses the existing scheduled-export engine (ScheduledDebugExport) rather than its own
// scheduler, so a redacted bundle drops each morning while the mode runs; the day count is derived from
// TestCentre.startedAt(.battery) against the registry default, so there is no extra persisted clock to
// drift. The pure status formatter is split out so it is unit-testable without a live clock. No em-dashes.
@MainActor
enum BatteryGuidedCapture {

    /// Arm the guided window: enable the daily scheduled export so the Battery bundle drops each morning
    /// for the duration. ScheduledDebugExport owns the BGTask / macOS timer; we only flip it on.
    static func arm() {
        ScheduledDebugExport.setEnabled(true)
    }

    /// Disarm when the user ends the Battery mode. The Test Centre Battery row owns this flag while the
    /// mode is on; flipping it off here is the days-twin of ending a Sleep capture.
    static func disarm() {
        ScheduledDebugExport.setEnabled(false)
    }

    /// "Capturing day K of N" status for the Test Centre Battery row. K is full-days-elapsed + 1, capped
    /// at N; once N whole days have passed it reads complete. Pure (now injected) so it is testable.
    static func statusText(startedAt: Date?, target: Int, now: Date = Date()) -> String {
        guard let startedAt else { return "Not started" }
        let elapsedDays = Int(now.timeIntervalSince(startedAt) / 86400)
        if elapsedDays >= target { return "Capture complete, \(target) of \(target) days" }
        return "Capturing day \(elapsedDays + 1) of \(target)"
    }

    /// Convenience binding for the Battery row: reads the live started-at + the registry default count.
    static func currentStatus(now: Date = Date()) -> String {
        let target: Int = {
            if case .guided(_, let count)? = TestModeRegistry.mode(.battery)?.capture { return count }
            return 3
        }()
        return statusText(startedAt: TestCentre.startedAt(.battery), target: target, now: now)
    }
}
